commit f502aad17b56bacb51b284b06a4821f2cc2ff6ed
Author: Kalesh Singh <kaleshsingh@google.com>
Date:   Sun Oct 13 13:28:26 2024 -0700

    ltp: controllers: memcg: Fix fault granularity
    
    Android emulates a userspace 16KiB page size on x86_64 emulators.
    The kernel however remains 4KiB and page faults are therefore handled
    in 4KiB granulatiry.
    
    Fix the memcg tests to use the kernel page size when faulting in pages.
    
    These changes are effectively no-ops for all targets except Android's
    x86_64 16KiB emulators
    
    Bug: 366098040
    Test: atest vts_ltp_test_x86_64:controllers.memcg_control_64bit#controllers.memcg_control_64bit
    Change-Id: I38d1e8f4f6aaf8d0fae6f2638da7ba71ad8ed1e4
    Signed-off-by: Kalesh Singh <kaleshsingh@google.com>

diff --git a/testcases/kernel/controllers/memcg/control/mem_process.c b/testcases/kernel/controllers/memcg/control/mem_process.c
index 8ecabb272..76cdf8066 100644
--- a/testcases/kernel/controllers/memcg/control/mem_process.c
+++ b/testcases/kernel/controllers/memcg/control/mem_process.c
@@ -11,6 +11,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include "pgsize_helpers.h"
 
 /*
  * Named pipe to act as a communication channel between
@@ -55,7 +56,7 @@ void touch_memory(char *p)
 	int i;
 	int pagesize = getpagesize();
 
-	for (i = 0; i < (int)memsize; i += pagesize)
+	for (i = 0; i < (int)memsize; i += kernel_page_size())
 		p[i] = 0xef;
 }
 
diff --git a/testcases/kernel/controllers/memcg/memcontrol02.c b/testcases/kernel/controllers/memcg/memcontrol02.c
index 0d93abd9e..5734c6dff 100644
--- a/testcases/kernel/controllers/memcg/memcontrol02.c
+++ b/testcases/kernel/controllers/memcg/memcontrol02.c
@@ -24,6 +24,7 @@
 #define _GNU_SOURCE
 
 #include "memcontrol_common.h"
+#include "pgsize_helpers.h"
 
 static size_t page_size;
 static struct tst_cg_group *cg_child;
@@ -39,7 +40,7 @@ static void alloc_anon_50M_check(void)
 		TST_CG_VER_IS_V1(tst_cg, "memory") ? "rss %zd" : "anon %zd";
 
 	buf = SAFE_MALLOC(size);
-	for (ptr = buf; ptr < buf + size; ptr += page_size)
+	for (ptr = buf; ptr < buf + size; ptr += kernel_page_size())
 		*ptr = 0;
 
 	SAFE_CG_SCANF(cg_child, "memory.current", "%zd", &current);
